src/common/history.c

```diff
@@ -1629,27 +1629,39 @@ void dt_history_hash_read(const int32_t imgid, dt_history_hash_values_t *hash)
     hash->basic_len = sqlite3_column_bytes(stmt, 0);
     if(buf)
     {
-      hash->basic = malloc(hash->basic_len);
+      hash->basic = g_malloc(hash->basic_len);
       memcpy(hash->basic, buf, hash->basic_len);
     }
     buf = (void *)sqlite3_column_blob(stmt, 1);
     hash->auto_apply_len = sqlite3_column_bytes(stmt, 1);
     if(buf)
     {
-      hash->auto_apply = malloc(hash->auto_apply_len);
+      hash->auto_apply = g_malloc(hash->auto_apply_len);
       memcpy(hash->auto_apply, buf, hash->auto_apply_len);
     }
     buf = (void *)sqlite3_column_blob(stmt, 2);
     hash->current_len = sqlite3_column_bytes(stmt, 2);
     if(buf)
     {
-      hash->current = malloc(hash->current_len);
+      hash->current = g_malloc(hash->current_len);
       memcpy(hash->current, buf, hash->current_len);
     }
   }
   sqlite3_finalize(stmt);
 }
 
+void dt_history_hash_free(dt_history_hash_values_t *hash)
+{
+  g_free(hash->basic);
+  hash->basic = NULL;
```

Comment:
Maybe overkill, but you could set `hash->xxx_len = 0` too...

Reply:
Not overkill, a good point. Thanks.